src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -1279,7 +1284,9 @@ export class NotebookModel extends Disposable implements INotebookModel {
 
 	private async updateKernelInfoOnKernelChange(kernel: nb.IKernel, kernelAlias?: string) {
 		await this.updateKernelInfo(kernel);
-		kernelAlias = this.kernelAliases.find(kernel => this._defaultLanguageInfo?.name === kernel.toLowerCase()) ?? kernelAlias;
+		if (!kernelAlias) {
+			kernelAlias = this.kernelAliases.find(k => this._defaultLanguageInfo?.name === k.toLowerCase());
```

Comment:
This defaultLanguageInfo gets updated further down, so when changing from Kusto to LogAnalytics we would see the old kusto language state and overwrite kernelAlias to be Kusto, since Kusto is also in the aliases list here. Then we'd have a different alias to the kernel display name LogAnalytics right after this, causing the kernelAlias to be set to undefined, which causes the notebook kernel to be set to the underlying SQL kernel as a fallback.